cpp/src/glm/qn/glm_logistic.cuh

```diff
@@ -41,6 +43,11 @@ struct LogisticLoss : GLMBase<T, LogisticLoss<T>> {
   }
 
   inline __device__ T dlz(const T y, const T z) const {
+    if (z < 0) {
```

Comment:
same as above. Explicit grouping of `myExp` call to avoid the extent of control divergence.

Reply:
Same as above.

Reply:
I have grouped the myExp calls to decrease control divergence.